### DISCUSSION TITLE
Scalene on Windows?

### DISCUSSION THREAD

**Discussion by waszil** [plasma-umass/scalene#299] [General]
Hi, is it planned to release Scalene for Windows? (I don't mean WSL, I mean "real" win(10)). It would be great! Thanks!

**Comment by emeryberger** [comment]
(edited)

Scalene does work on Windows! However, only some of Scalene's functionality is currently supported on Windows (CPU profiling, including separating out Python/native/system time, at a line granularity). We plan to do more (supporting memory profiling, etc.) but unfortunately, the Windows environment is not as friendly to Scalene as Mac or Linux, so it will take some work. If anyone is interested in helping, we know what to do in broad strokes and would be happy to work with someone external on this.

**Reply by emeryberger** [reply]
I should have noted: Scalene _does_ work on Windows! However, only some of Scalene's functionality is currently supported on Windows (CPU profiling, including separating out Python/native/system time, at a line granularity).

**Reply by danielhrisca** [reply]
Hello, can you confirm if multiprocessing is supported for Windows? (doesn't seem to work on my setup)

**Reply by waszil** [reply]
Well, on my setup it does not work at all, either. I have Windows 10, Python 3.8.5, and on a clean venv, it fails to print even the help message:

```
(venv) C:\work\scalene_test>scalene --help
Traceback (most recent call last):
  File "c:\Anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\Anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\work\scalene_test\venv\Scripts\scalene.exe\__main__.py", line 4, in <module>
  File "C:\work\scalene_test\venv\lib\site-packages\scalene\__main__.py", line 4, in <module>
    from scalene import scalene_profiler
  File "C:\work\scalene_test\venv\lib\site-packages\scalene\scalene_profiler.py", line 30, in <module>
    import resource
ModuleNotFoundError: No module named 'resource'
```

How and what exactly can be used on Windows right now?

**Reply by danielhrisca** [reply]
I think you need the latest github code to avoid the error

**Reply by waszil** [reply]
You're right, with latest master, it works. Thanks.

**Reply by emeryberger** [reply]
This is also fixed in the latest release (1.4.0).
